t/02-test_ocr.t

```diff
@@ -8,47 +8,124 @@ use FindBin '$Bin';
+    {
+        no warnings qw(redefine prototype);
+        no strict 'refs';
+        my $saved_func = \&tinycv::Image::write;
+        *{'tinycv::Image::write'} = sub { return undef; };
+        stderr_like(
+            sub { is(img_to_str($img_area), undef, 'Check undef if writing img failed'); },
+            qr/.*Writing image to prepare for OCR failed!.*/,
+            'Check stderr for failed writing of img.');
+        *{'tinycv::Image::write'} = $saved_func;
+    }
 
-my $ocr;
-for my $area (@{$res->{needle}->{area}}) {
-    next unless $area->{type} eq 'ocr';
-    $ocr .= ocr::tesseract($img1, $area);
-}
+  SKIP: {
+        skip('The OCR program is not installed.', 1)
+          unless (ocr_installed());
 
-ok defined $ocr, 'OCR area found' and
-  ok($ocr =~ /Memory Test.*Video Mode/s, 'multiple OCR regions');
+        # Some checks that require ugly overrides.
+        {
+            my $fakepath = tempdir();
+            my $fakeocr = catfile($fakepath, 'gocr');
+            my $fakeocr_content = encode('UTF-8', "#!/usr/bin/env sh\necho 'I am fake' 1>&2\nexit 1");
+            open(my $fakeocr_fh, '>', $fakeocr);
 
-done_testing;
+            # print seems to be swallowed by the test framework
+            syswrite($fakeocr_fh, $fakeocr_content, length($fakeocr_content));
+
+            close($fakeocr_fh);
+            chmod(0755, $fakeocr);
+            my $path_sav = $ENV{PATH};
+            $ENV{PATH} = $fakepath . ':' . $ENV{PATH};
+            stderr_like(
+                sub { is(img_to_str($img_area), undef, 'Check undef if OCR program has failed.'); },
+                qr/.*The OCR program exited with error.*/,
+                'Check stderr for failed OCR program.');
+            $ENV{PATH} = $path_sav;
+        }
+
+        $xpos = 0;
+        $ypos = 254;
+        $width = 1025;
+        $height = 17;
+        $img_area = $img->copyrect($xpos, $ypos, $width, $height);
+        stderr_like(
+            sub { is(img_to_str($img_area), undef, 'Check undef for exceeding horizontal pixel count'); },
+            qr/.*Skipping OCR evaluation of image, resolution too high\(max\..*/,
+            'Check stderr for exceeding horizontal pixel count');
 
+        $xpos = 256;
+        $ypos = 0;
+        $width = 194;
+        $height = 769;
+        $img_area = $img->copyrect($xpos, $ypos, $width, $height);
+        stderr_like(
+            sub { is(img_to_str($img_area), undef, 'Check undef for exceeding vertical pixel count'); },
+            qr/.*Skipping OCR evaluation of image, resolution too high\(max\..*/,
+            'Check stderr for exceeding vertical pixel count');
+
+        $img = tinycv::read(dirname(__FILE__) . '/data' . '/bootmenu.test.png');
+        $xpos = 256;
+        $ypos = 254;
+        $width = 194;
+        $height = 17;
+        $img_area = $img->copyrect($xpos, $ypos, $width, $height);
+        $ocr_str = img_to_str($img_area);
+        diag("Found OCR String:\n$ocr_str");
```

Comment:
Please make sure that in non-verbose mode the tests do not output anything, see https://github.com/os-autoinst/os-autoinst/actions/runs/6347763302/job/17243248731?pr=2276#step:3:635

You can use `note` here. And for all other output I suggest to use Test::Output as you already do in other cases